extension/onboarding/onboardingController.jsx

```diff
@@ -0,0 +1,69 @@
+/* globals React, ReactDOM, settings */
+
+this.onboardingController = (function() {
+  const exports = {};
+  const { useState, useEffect } = React;
+  const onboardingContainer = document.getElementById("onboarding-container");
+  let isInitialized = false;
+  let userSettings;
+
+  exports.OnboardingController = function() {
+    const [optinViewAlreadyShown, setOptinViewShown] = useState(false);
+    const [permissionError, setPermissionError] = useState(null);
+
+    useEffect(() => {
+      if (!isInitialized) {
+        isInitialized = true;
+        init();
+      }
+    });
+
+    const init = async () => {
+      if (location.pathname.endsWith("onboard.html")) {
+        launchPermission();
+      }
+      const result = await settings.getSettingsAndOptions();
+      userSettings = result.settings;
+    };
+
+    const setOptinValue = async value => {
+      // TODO: Is this the right setting to update?
+      userSettings.collectAudio = value;
+      await settings.saveSettings(userSettings);
+    };
+
+    const launchPermission = async () => {
+      try {
```

Comment:
Potentially right here could be `setPermissionError("Waiting")` (or something like that, where we say that we're waiting for the user to enable the microphone)